src/citrine/resources/ara_job.py

```diff
@@ -0,0 +1,179 @@
+from typing import List, Union, Optional, Set
+from uuid import UUID
+from uuid import uuid4
+
+from citrine._serialization.properties import Set as PropertySet, String, Property
+from citrine._rest.resource import Resource
+from citrine._session import Session
+from citrine.resources.api_error import ApiError
+from citrine.resources.ara_definition import AraDefinition
+from citrine.resources.project import Project
+from citrine._serialization import properties
+
+
+class AraJobSubmissionResponse(Resource['AraJobStatus']):
+    """
+    [ALPHA] a response to a submit-job request for the job submission framework
+
+    Parameters
+    ----------
+    job_id: UUID
+        job id of the of a job submission request
+    """
+    def __init__(self, job_id: UUID):
+        self.job_id = job_id
+
+
+class TaskNode(Resource['TaskNode']):
+    """
+    [ALPHA] individual task status
+
+    Parameters
+    ----------
+    id: str
+        unique identification number for the job task.
+    task_type: str
+        the type of task running
+    status: str
+        the last reported status of this particular task.
+    dependencies: Set[str]
+        all the tasks that this task is dependent on.
+    failure_reason: Optional[str]
+        if a task has failed, the failure reason will be in this parameter.
+    """
+    id = properties.String("id")
+    task_type = properties.String("task_type")
+    status = properties.String("status")
+    dependencies = PropertySet(String(), "dependencies")
+    failure_reason = properties.Optional(String(), "failure_reason")
+
+    def __init__(
+            self,
+            id: str,
+            task_type: str,
+            status: str,
+            dependencies: Set[str],
+            failure_reason: Optional[str]
+    ):
+        self.id = id
+        self.task_type = task_type
+        self.status = status
+        self.dependencies = dependencies
+        self.failure_reason = failure_reason
+
+
+class TaskNodeType(Property[TaskNode, dict]):
+    @property
+    def underlying_types(self):
+        return TaskNode
+
+    @property
+    def serialized_types(self):
+        return dict
+
+    def _serialize(self, value: TaskNode) -> dict:
+        result = {"id": value.id, "task_type": value.task_type, "status": value.status,
+                  "failure_reason": value.failure_reason, "dependencies": value.dependencies}
+        return result
+
+    def _deserialize(self, value: dict) -> TaskNode:
+        return TaskNode.build(value)
```

Comment:
You shouldn't need to define a new `TaskNode` Property. By virtue of being a `Resource` it should do this automatically through the code in `properties.Object`

Reply:
it didn't work, but cause there are two paths for serialization (jobStatus has a list of tasknodes). I had to create the type to make it work

Reply:
You can chain them together to specify a list of TaskNodes. Example: https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/ara/rows.py#L54